src/main/java/com/laytonsmith/core/functions/Threading.java

```diff
@@ -575,15 +577,197 @@ public String getName() {
-					+ " InterruptedException is thrown. If the id is unknown, an IllegalArgumentException is thrown.";
+					+ " (Sending 0 for this value causes an infinite wait.) If the timeout occurs or thread interrupted, an"
+					+ " InterruptedException is thrown.";
+		}
+
+		@Override
+		public Version since() {
+			return MSVersion.V3_3_4;
+		}
+
+	}
+
+	@api
+	public static class x_thread_is_alive extends AbstractFunction {
+
+		@Override
+		public Class<? extends CREThrowable>[] thrown() {
+			return new Class[0];
+		}
+
+		@Override
+		public boolean isRestricted() {
+			return true;
+		}
+
+		@Override
+		public Boolean runAsync() {
+			return null;
+		}
+
+		@Override
+		public Mixed exec(Target t, Environment environment, Mixed... args) throws ConfigRuntimeException {
+			String id = args[0].val();
+			Thread th = THREAD_ID_MAP.get(id);
+			if(th == null) {
+				return CBoolean.FALSE;
+			}
+			return CBoolean.get(th.isAlive());
+		}
+
+		@Override
+		public Version since() {
+			return MSVersion.V3_3_4;
+		}
+
+		@Override
+		public String getName() {
+			return "x_thread_is_alive";
+		}
+
+		@Override
+		public Integer[] numArgs() {
+			return new Integer[]{1};
+		}
+
+		@Override
+		public String docs() {
+			return "bool {mixed thread} Tests if this thread is alive. A thread is alive if it has been started and has not yet died.";
+		}
+	}
+
+	@api
+	@seealso({x_is_interrupted.class})
+	public static class x_interrupt extends AbstractFunction {
+
+		@Override
+		public Class<? extends CREThrowable>[] thrown() {
+			return new Class[0];
+		}
+
+		@Override
+		public boolean isRestricted() {
+			return true;
+		}
+
+		@Override
+		public Boolean runAsync() {
+			return null;
+		}
+
+		@Override
+		public Mixed exec(Target t, Environment environment, Mixed... args) throws ConfigRuntimeException {
+			String id = args[0].val();
+			Thread th = THREAD_ID_MAP.get(id);
+			if(th != null) {
+				th.interrupt();
+			}
+			return CVoid.VOID;
 		}
 
 		@Override
 		public Version since() {
 			return MSVersion.V3_3_4;
 		}
 
+		@Override
+		public String getName() {
+			return "x_interrupt";
+		}
+
+		@Override
+		public Integer[] numArgs() {
+			return new Integer[]{1};
+		}
+
+		@Override
+		public String docs() {
+			return "void {mixed thread} Interrupts this thread. In other words \"interrupted status\" set true."
+					+ " The \"interrupted status\" can be checked using the method: " + x_is_interrupted.class.getSimpleName() + "()"
+					+ " If this thread is blocked in an invocation of the " + Scheduling.sleep.class.getSimpleName() + "(), " + x_thread_join.class.getSimpleName() + "() methods"
+					+ " then its \"interrupt status\" will be cleared and it will receive an InterruptedException.";
+		}
 	}
 
+	@api
+	@seealso({x_interrupt.class})
+	public static class x_is_interrupted extends AbstractFunction {
+
+		private static java.lang.reflect.Method interrupt;
```

Comment:
It is questionable whether `x_is_interrupted(not_current_thread, true)` should be allowed or not. The method you're accessing through reflection to implement this is an unsynchronized native method that might not be thread-safe. For `x_is_interrupted(current_thread, true)`, you can use static `Thread.interrupted()`, which clears the flag.

Reply:
Then it is impossible to implement the case x_is_interrupted (other_thread, true).

`interrupted()` and `isInterrupted()` methods do not synchronize native `isInterrupted(clearFlag)`

Reply:
That's why I'm proposing to not allow that case. The `isInterrupted(true)` call in `Thread` is implicitly synchronized, as it can only be called from the current thread.

Reply:
Then i need to do the synchronization block on the `th`

Reply:
I actually think that `x_is_interrupted(not_current_thread, true)` should not be allowed since when a thread is interrupted, then that thread should deal with it. For me it doesn't make sense to have a thread clear the interrupt status of another thread. Lets also throw this at @LadyCailin to see what she has to say on this.